packages/tss/src/tss.ts

```diff
@@ -0,0 +1,677 @@
+}
+
+export interface TKeyTSSInitArgs extends TKeyInitArgs {
+  deviceTSSShare?: BN;
+  deviceTSSIndex?: number;
+  factorPub?: Point;
+}
+
+export class TKeyTSS extends ThresholdKey {
+  serviceProvider: TSSTorusServiceProvider = null;
+
+  private _tssKeyType: KeyType;
+
+  private _tssCurve: EC;
+
+  private _tssTag: string;
+
+  private _accountSalt: string;
+
+  constructor(args?: TSSTKeyArgs) {
+    super(args);
+    const { serviceProvider, storageLayer, tssTag = "default", tssKeyType } = args;
+    this.serviceProvider = serviceProvider;
+    this.storageLayer = storageLayer;
+    this._tssTag = tssTag;
+    this._tssKeyType = tssKeyType;
+    this._tssCurve = new EC(tssKeyType);
+  }
+
+  public get tssTag(): string {
+    return this._tssTag;
+  }
+
+  public get tssKeyType(): KeyType {
+    return this._tssKeyType;
+  }
+
+  public set tssTag(tag: string) {
+    this._tssTag = tag;
+  }
+
+  async initialize(params?: TKeyTSSInitArgs): Promise<KeyDetails> {
+    const keyDetails = await super.initialize(params);
+
+    if (!this.metadata.tssPolyCommits[this.tssTag]) {
+      // if tss shares have not been created for this tssTag, create new tss sharing
+      const { factorEncs, factorPubs, tssPolyCommits } = await this._initializeNewTSSKey(
+        this.tssTag,
+        params.deviceTSSShare,
+        params.factorPub,
+        params.deviceTSSIndex
+      );
+      this.metadata.addTSSData({ tssKeyType: this.tssKeyType, tssTag: this.tssTag, tssNonce: 0, tssPolyCommits, factorPubs, factorEncs });
+      const accountSalt = generateSalt(this._tssCurve);
+      await this._setTKeyStoreItem(TSS_MODULE, {
+        id: "accountSalt",
+        value: accountSalt,
+      } as IAccountSaltStore);
+      this._accountSalt = accountSalt;
+    }
+    return keyDetails;
+  }
+
+  /**
+   * Returns the encrypted data associated with the given factor public key.
+   */
+  getFactorEncs(factorPub: Point): FactorEnc {
+    if (!this.metadata) throw CoreError.metadataUndefined();
+    if (!this.metadata.factorEncs) throw CoreError.default("no factor encs mapping");
+    if (!this.metadata.factorPubs) throw CoreError.default("no factor pubs mapping");
+    const factorPubs = this.metadata.factorPubs[this.tssTag];
+    if (!factorPubs) throw CoreError.default(`no factor pubs for this tssTag ${this.tssTag}`);
+    if (factorPubs.filter((f) => f.x.cmp(factorPub.x) === 0 && f.y.cmp(factorPub.y) === 0).length === 0)
+      throw CoreError.default(`factor pub ${factorPub} not found for tssTag ${this.tssTag}`);
+    if (!this.metadata.factorEncs[this.tssTag]) throw CoreError.default(`no factor encs for tssTag ${this.tssTag}`);
+    const factorPubID = factorPub.x.toString(16, 64);
+    return this.metadata.factorEncs[this.tssTag][factorPubID];
+  }
+
+  /**
+   * Returns the TSS share associated with the given factor private key.
+   */
+  async getTSSShare(
+    factorKey: BN,
+    opts?: {
+      threshold?: number;
+      accountIndex?: number;
```

Comment:
as per last discussion, we would fix the account index in this pr.
just note here for tracking

Reply:
It's already implemented.
https://github.com/tkey/tkey/pull/268/commits/ed69263c11437a6f8ee5fb3462f73872732f7874
You can now supply coefficient for share derivation.